### PR TITLE
Laser Cannon Changes

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Inspired by the "help how do we balance xenos when lasercannons kill them in two shots" discussion.

Laser Cannon is now the Accelerator Laser Cannon. The projectile grows the farther it flies. There is no cap to to this damage, but it starts off very weak.

This should make it a more situational weapon/sidegrade rather than X2 DAMAGE GUN.

For those who want to crunch numbers, its 5 damage to start off and +7 damage per tile.

:cl: Kor
add: The Laser Cannon in RD has been replaced with the Accelerator Laser Cannon. Try it out!
/:cl:

### CODE REVIEW

**Comment by duncathan** [comment]
this is gonna be fun on dream

File: code/modules/projectiles/guns/energy/laser.dm [hunk 1]
```diff
@@ -67,7 +67,7 @@
 	flags =  CONDUCT
 	slot_flags = SLOT_BACK
 	origin_tech = "combat=4;materials=3;powerstorage=3"
-	ammo_type = list(/obj/item/ammo_casing/energy/laser/heavy)
+	ammo_type = list(/obj/item/ammo_casing/energy/laser/accelerator)
```
Comment (Iamgoofball) [inline code/modules/projectiles/guns/energy/laser.dm]: I think you forgot the code for the projectile itself fam :ok_hand:

**Comment by Lumipharon** [comment]
Isnt the more logical solution to not make xenos utterly wrecked by burn damage?

**Comment by KorPhaeron** [comment]
@Lumipharon 

I guess that's also possible, but as OOC said

OOC: Sekas: laser cannon should be different than just 'lol its a laser gune but it do more hurt'

I think it'll be more interesting this way.

**Comment by Iamgoofball** [comment]
Before you downvote because muh science nerfs

Imagine firing this down escape hallway

File: code/modules/projectiles/guns/energy/laser.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/projectiles/guns/energy/laser.dm]: How does it compile then? Checkmate athiests

Comment (GunHog) [inline code/modules/projectiles/guns/energy/laser.dm]: You did not test this? At least provide a damage graph so we can tell how big of a nerf/buff this is. I am not interested in your "infinite damage" memes, I want to know what this thing will realistically do in most situations.

**Comment by Core0verload** [comment]
Xenos getting raped by laser cannons is okay. That's how it MUST be if they didn't sabotaged the RnD.

One barely visible alien hunter can easily pop out of scrubber and smash the console, destroying all the research data for dumbass scientists who didn't synced, or pausing the production for the ones who did.

An alien with acid split can invade the toxins storage, release the O2 and PL, then acid the door, leaving the scientists with a huge toxic flammable mess in the hallway. Then a single spark will burn the fuck out of it and make all the sci completely unusable, at least without a space suit.

**Comment by Ninja137** [comment]
>Fire
>Space
>A room getting exposed

>Any of these being legitimate threats

Engineering repairs it 100% in less than 5 minutes, if that. It's entirely irrelevant.

**Comment by KorPhaeron** [comment]
>complain I use "the nuclear" option on stuff
>go out of my way to design a gun with a unique playstyle that has potential to be even stronger than the old version
>still complain

**Comment by Core0verload** [comment]
It's a little hard to do repairs when xenos are around. They hit hard in melee and are completely immune to engineers' stunprods. And depressurization makes them immune to shit-grade flamethrowers too, making repair crew a nice snack for them. Because hunters have stealth that works really fucking good on both weed and burned floors.

**Comment by Core0verload** [comment]
@KorPhaeron, maybe you should just stop trying to balance anything, given the amount of negative feedback every your attempt receives.

File: code/modules/projectiles/guns/energy/laser.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/projectiles/guns/energy/laser.dm]: It's been in the game for literal months GunHog, it's already been tested.

**Comment by PKPenguin321** [comment]
This is fucking dope

**Comment by Lumipharon** [comment]
I'm perfectly fine with this btw, just pointing out that this doesn't actually fix the problem.

**Comment by KorPhaeron** [comment]
@Lumipharon 

yeah no totally xenos themselves need rebalancing, the whole conversation just made me remember how dumb having a gun with the only difference being double damage is

**Comment by Steelpoint** [comment]
So after about five tiles of travel time the projectile will be dealing 40 damage again.

I'm not against this change, it encourages the user to keep their distance but also encourages their attacker to close the range.

But it also means someone firing down a corridor is suddenly going to be dealing 100+ damage shots at long range, if someone is unlucky they may well get two shot KILLED if they're standing in the wrong place at the wrong time.

**Comment by KorPhaeron** [comment]
>So after about five tiles of travel time the projectile will be dealing 40 damage again

Yep. So it's only very slightly worse than a lasercannon at midrange. It gains massive potential past that point though. 

>if someone is unlucky they may well get two shot KILLED if they're standing in the wrong place at the wrong time.

Also yes. I think the tradeoff of being weak at melee range is well worth it for the long range potential. It might even be too strong like this.

**Comment by Bawhoppen** [comment]
I don't like this mechanic, do something else please

**Comment by Robustin** [comment]
Looks good. Now merge #16164 so I can close this generation of PR's and get on with a new generation of memes.

File: code/modules/projectiles/guns/projectile/sniper.dm [hunk 2]
```diff
@@ -188,3 +188,7 @@
 		breakthings = TRUE
 	else if(damage > 25)
 		icon_state = "gauss"
+
+/obj/item/projectile/bullet/sniper/accelerator/laser
```
Comment (GunHog) [inline code/modules/projectiles/guns/projectile/sniper.dm]: So this "laser" is actually just a bullet that comes from a laser cannon? Did you test this to ensure that it:
- Deals burn damage, like a laser does?
- Is blocked or has damage reduced according to laser armor, and not bullet?
- Has the correct sprite and sound?

**Comment by CPTANT** [comment]
I think Heavy lasers spontaneously setting you on fire was a more interesting change than this.

**Comment by PKPenguin321** [comment]
We should really make something that's like an emitter but functions like these guns

File: code/modules/projectiles/guns/projectile/sniper.dm [hunk 3]
```diff
@@ -173,7 +173,7 @@
 	icon_state = ".50"
 
 /obj/item/projectile/bullet/sniper/accelerator
-	icon_state = "gaussweak"
+	icon_state = "scatterlaser"
```
Comment (ChangelingRain) [inline code/modules/projectiles/guns/projectile/sniper.dm]: KOR WHY

**Comment by optimumtact** [comment]
This needs a cap

File: code/modules/projectiles/guns/projectile/sniper.dm (hunk 3 above)
Comment (GunHog) [inline code/modules/projectiles/guns/projectile/sniper.dm]: This is the icon state for the _projectile sniper rifle round_, not your new subtype. Did you check that the heavy laser sound is preserved as well? (Unless you do not intend to keep it?)

File: code/modules/projectiles/guns/projectile/sniper.dm [hunk 4]
```diff
@@ -182,13 +182,12 @@
 
 /obj/item/projectile/bullet/sniper/accelerator/Range()
 	..()
-	damage += 5
-	if(damage > 40)
-		icon_state = "gaussstrong"
+	damage += 7
```
Comment (GunHog) [inline code/modules/projectiles/guns/projectile/sniper.dm]: This is also affecting the balance of the original projectile, file from the nuke op sniper rifle. Was this your intent?

File: code/modules/projectiles/guns/projectile/sniper.dm [hunk 5]
```diff
@@ -182,13 +182,12 @@
...
+	transform *= TransformUsingVariable(1.2 , 100)
```
Comment (GunHog) [inline code/modules/projectiles/guns/projectile/sniper.dm]: So, it gets bigger as it travels. Cool beans!

File: code/modules/projectiles/guns/projectile/sniper.dm [hunk 6]
```diff
@@ -182,9 +182,12 @@
...
+	if(damage > 42)
 		breakthings = TRUE
-	else if(damage > 25)
-		icon_state = "gauss"
+
+/obj/item/projectile/bullet/sniper/accelerator/laser
```
Comment (GunHog) [inline code/modules/projectiles/guns/projectile/sniper.dm]: Agreed. You would not have to check for so many things in your testing this way. Either you are going to be copy-pasting the bullet's unique code to the laser, or copy-pasting the laser's properties to a projectile. It is better to keep the subtypes consistent in this case.

Of course, if you were ultra obsessive, your third option is to elevate the accelerator properties such that they can be used by both physical and energy projectiles. I do not recommend it, however.

File: code/modules/projectiles/guns/projectile/sniper.dm (hunk 4 above)
Comment (KorPhaeron) [inline code/modules/projectiles/guns/projectile/sniper.dm]: http://www.ss13.eu/tgdb/tg/latest_stats.html#uplinkitems

People only buy the penetrator rounds anyway

File: code/modules/projectiles/guns/projectile/sniper.dm (hunk 6 above)
Comment (KorPhaeron) [inline code/modules/projectiles/guns/projectile/sniper.dm]: Maybe I'll just axe the accelerator bullet since nobody buys that

Comment (GunHog) [inline code/modules/projectiles/guns/projectile/sniper.dm]: Eh, fair enough.

**Comment by Dorsisdwarf** [comment]
Damage increase needs to grow slower, like 40 damage at 7, 100 damage at 20 tiles.

**Comment by Supermichael777** [comment]
shooting 20 tiles without a scope, right

**Comment by Dorsisdwarf** [comment]
How far is the bar to the escape corridor on box?

File: code/modules/projectiles/guns/projectile/sniper.dm (hunk 6 above)
Comment (duncathan) [inline code/modules/projectiles/guns/projectile/sniper.dm]: @KorPhaeron

Comment (KorPhaeron) [inline code/modules/projectiles/guns/projectile/sniper.dm]: Soon(TM)

**Comment by Saegrimr** [comment]
Can't wait to go inna space and fire at the windows in escape.

**Comment by KorPhaeron** [comment]
@duncathan @ChangelingRain 

I finished/tested it.

File: code/modules/projectiles/guns/energy/laser.dm [hunk 7]
```diff
@@ -67,10 +69,26 @@
...
 	ammo_x_offset = 3
 
+/obj/item/ammo_casing/energy/laser/accelerator
+	projectile_type = /obj/item/projectile/beam/laser/accelerator
+	select_name = "accelerator"
+	fire_sound = 'sound/weapons/lasercannonfire.ogg'
+
+/obj/item/projectile/beam/laser/accelerator
+	name = "accelerator laser"
+	icon_state = "scatterlaser"
```
Comment (ChangelingRain) [inline code/modules/projectiles/guns/energy/laser.dm]: use the laser icon state

File: code/modules/uplink/uplink_item.dm [hunk 8]
```diff
@@ -452,12 +452,6 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
 	item = /obj/item/ammo_box/magazine/sniper_rounds/penetrator
 	cost = 5
 
-/datum/uplink_item/ammo/sniper/accelerator
```
Comment (ChangelingRain) [inline code/modules/uplink/uplink_item.dm]: Why specifically are you removing this

File: code/modules/projectiles/guns/energy/laser.dm (hunk 7 above)
Comment (KorPhaeron) [inline code/modules/projectiles/guns/energy/laser.dm]: It starts out weaker/smaller than a laser though

File: code/modules/uplink/uplink_item.dm (hunk 8 above)
Comment (KorPhaeron) [inline code/modules/uplink/uplink_item.dm]: http://www.ss13.eu/tgdb/tg/latest_stats.html#uplinkitems

Literally never bought, and I want to use the same code for this

File: code/modules/projectiles/guns/energy/laser.dm (hunk 7 above)
Comment (ChangelingRain) [inline code/modules/projectiles/guns/energy/laser.dm]: use the laser icon state the scatter laser icon state will look really weird and out of place with this

Comment (KorPhaeron) [inline code/modules/projectiles/guns/energy/laser.dm]: I just tested with the scatter one though, it looks good having a little circle grow gigantic as it flies

Comment (KorPhaeron) [inline code/modules/projectiles/guns/energy/laser.dm]: Cheridan> FOR THE RECORD I PREFER THE CIRCLE

Comment (ChangelingRain) [inline code/modules/projectiles/guns/energy/laser.dm]: I DISAGREE WITH CIRCLES

Comment (KorPhaeron) [inline code/modules/projectiles/guns/energy/laser.dm]: Cheridan is our king though

**Pull request merged**